### ISSUE TITLE
Stackage?

### ISSUE THREAD

**Issue by janpath** [LeventErkok/sbv#422]
Can we get this back on Stackage? As far as I can tell it was removed, because crackNum failed compiling, but that seems not to be the case anymore (at least just adding everything to extra-deps worked without problems).

**Comment by LeventErkok** [comment]
Yes; that would be nice. I did get it into stackage about a year ago (I think), but then it got dropped and I just can't quite work out all the stackage requirements. If you can see it through, it'd be much appreciated.

**Comment by LeventErkok** [comment]
@janpath 

I tried:

   https://github.com/commercialhaskell/stackage/pull/4152

But it's choking, complaining about something to do with `fuzzyset`; which I've no idea where it's coming from.

This is precisely the reason why I stopped caring about stackage in the past, as it's impossible to figure out why things fail. Any help would be appreciated.

**Comment by janpath** [comment]
Thank you. The error does not seem to be related to sbv or crackNum. Stackage will always use the newest versions from Hackage and since a new version of `base-unicode-symbols` was uploaded today `fuzzyset`, which has a dependency on a specific version of `base-unicode-symbols` now fails. This is not your fault and does not indicate that anything is wrong with your packages.

**Comment by janpath** [comment]
I don't know if you already tested building the packages locally from Hackage, as the checklist in the commit message demands, but when I did, I needed cabal-install < 2 in my PATH, with newer versions it would fail.

**Comment by LeventErkok** [comment]
Cool. Do you know what we need to put into their .yaml file so we can get a build going there?

**Comment by janpath** [comment]
I do not think we have to do anything. When the Stackage curators build the next nightly they will probably put an upper bound on `base-unicode-symbols` and open an issue pinging the `fuzzyset` maintainers to fix their dependencies (see [here](https://github.com/commercialhaskell/stackage/blob/master/MAINTAINERS.md#following-dependency-upgrades)). If you just want to see, if the CI succeeds, I suppose you could do `base-unicode-symbols == 2.2.2.4` in `build-constraints.yaml`.

**Comment by janpath** [comment]
Now that the package is in nightly it will be included in the next LTS. Maybe you could also add it to LTS-12? You just have to open an issue on the [lts-haskell repo](https://github.com/fpco/lts-haskell/issues/new) (see [here](https://github.com/commercialhaskell/stackage/blob/master/MAINTAINERS.md#adding-a-package-to-an-lts-snapshot)).

**Comment by LeventErkok** [comment]
OK, filed an issue: https://github.com/commercialhaskell/lts-haskell/issues/185

Question: If I push a new version of SBV to Hackage (which I intend to do in the next week or so), does it automatically pick up the latest version from Hackage, or do I have to go through these steps again?

**Comment by janpath** [comment]
Nightly will automatically pick up the newest version from Hackage. LTS releases will only pick up new minor versions. However since you seem to be following SemVer instead of PVP I do not know how that is handled. In the [MAINTAINERS.md](https://github.com/commercialhaskell/stackage/blob/master/MAINTAINERS.md#lts-package-guarantees-and-exceptions) it says:

> * If a package does not follow the PVP in its version number policy, applying the standard version bump rules would not necessarily makes sense. As an example, suppose package foo decides to follow SemVer instead of the PVP. By our standard rules of version bumps, a change from foo-1.2.0 to foo-1.3.0 would be considered a major version bump, and disallowed in an LTS minor version bump. However, if a package is following SemVer, this would not be a breaking change, and curators may elect to include it.

But I do not know how they know what packages use SemVer and which use PVP, there doesn't seem to be anything about that in `build-constraints.yaml` and I can not imagining the Stackage curators manually going through all the packages, looking at the version numbers and guessing whether it follows SemVer or PVP. Maybe they will only include new SemVer minor versions if the maintainer opens an issue on lts-haskell?

**Comment by janpath** [comment]
I made an issue, asking how this is handled [here](https://github.com/commercialhaskell/stackage/issues/4163), as this does not seem to be documented anywhere.

**Comment by phadej** [comment]
@janpath, if you look at diffs http://hdiff.luite.com/cgit/sbv/diff?id=7.10&id2=7.9 then there are definitely breaking changes. (also 7.10 to 7.11 looked like it might break users, not sure about 7.11 to 7.12). So on the quick glance `sbv` doesn't follow SemVer. 

```diff
   -- ** Soft assumptions
   -- $softAssertions
-  , assertSoft , Penalty(..)
+  , assertWithPenalty , Penalty(..)
```

maybe that change doesn't

**Comment by janpath** [comment]
@phadej I see, thanks. I was just skimming the changelog.

**Comment by janpath** [comment]
Thank you.

Regarding SemVer vs PVP. I understand that SemVer may seem like the better alternative at first, it certainly did to me, however the PVP is really deeply ingrained in the Haskell ecosystem. I am not aware of any popular packages that do not use the PVP, tooling like Stackage assumes PVP and [so do many developers](https://github.com/commercialhaskell/stackage/issues/4163#issuecomment-440951122). I think the technical merits of PVP vs SemVer can be debated, but I do not think the differences are big enough to use the non-standard alternative.
